pkg/component/server/konnectivity.go

```diff
@@ -124,7 +161,97 @@ func (k *Konnectivity) writeKonnectivityAgent() error {
 	if err != nil {
 		return fmt.Errorf("failed to write konnectivity agent manifest: %v", err)
 	}
+	return nil
+}
+
+func (k *Konnectivity) runLease() {
+	k.leaseCtx, k.leaseCancel = context.WithCancel(context.Background())
+	go func(ctx context.Context) {
+		logrus.Infof("starting %v lease watcher", serviceName)
+		leaseLock, err := k.newLeaseLock()
+		if err != nil {
+			logrus.Error(err)
+		}
+		k.leaseLock = leaseLock
+		for {
+			select {
+			case <-ctx.Done():
+				logrus.Debugf("stopping lease watcher for %v", serviceName)
+				return
+			default:
+				k.leaseLock.LeaseRunner(context.Background())
+			}
+		}
+	}(k.leaseCtx)
+
+	go func(ctx context.Context) {
+		logrus.Infof("watching %v lease holders", serviceName)
+		ticker := time.NewTicker(30 * time.Second)
+
```

Comment:
```suggestion
defer ticker.Close()
```
Just to be sure we do not leak tickers when restarting konnectivity